FirebaseMLModelDownloader/Sources/ModelDownloader.swift

```diff
@@ -490,91 +475,131 @@ extension ModelDownloader {
         /// Local model info is the latest model info.
         case .notModified:
           guard let localModel = self.getLocalModel(modelName: modelName) else {
-            /// This can only happen if either local model info or the model file was suddenly wiped out in the middle of model info request and server response
-            // TODO: Consider handling: If model file is deleted after local model info is retrieved but before model info network call.
+            /// This can only happen if either local model info or the model file was wiped out after model info request but before server response.
             self
               .mainQueueHandler(completion(.failure(.internalError(description: ModelDownloader
-                  .ErrorDescription.deletedLocalModelInfo))))
+                  .ErrorDescription.deletedLocalModelInfoOrFile))))
             return
           }
-
           self.mainQueueHandler(completion(.success(localModel)))
         }
+      /// Error retrieving model info.
       case let .failure(error):
         self.mainQueueHandler(completion(.failure(error)))
       }
     }
   }
 }
 
-/// Model downloader extension for testing.
-extension ModelDownloader {
-  /// Model downloader instance for testing.
-  // TODO: Consider using protocols
-  static func modelDownloaderWithDefaults(_ defaults: UserDefaults,
-                                          app: FirebaseApp) -> ModelDownloader {
-    let downloader = ModelDownloader(app: app, defaults: defaults)
-    return downloader
-  }
+/// Possible errors with model downloading.
+public enum DownloadError: Error, Equatable {
+  /// No model with this name exists on server.
+  case notFound
+  /// Invalid, incomplete, or missing permissions for model download.
+  case permissionDenied
+  /// Conditions not met to perform download.
+  case failedPrecondition
+  /// Requests quota exhausted.
+  case resourceExhausted
+  /// Not enough space for model on device.
+  case notEnoughSpace
+  /// Malformed model name or Firebase app options.
+  case invalidArgument
+  /// Other errors with description.
+  case internalError(description: String)
 }
 
-/// Possible error messages while using model downloader.
+/// Possible errors with locating a model file on device.
+public enum DownloadedModelError: Error {
+  /// No model with this name exists on device.
+  case notFound
+  /// File system error.
+  case fileIOError(description: String)
+  /// Other errors with description.
+  case internalError(description: String)
+}
+
+/// Extension to handle internally meaningful errors.
+extension DownloadError {
+  /// Model download URL expired before model download, raised internally. Model info retrieval and download is retried `numberOfRetries` times before failing externally.
```

Comment:
Nit: this is likely too long, should be 100 characters max (I think? Could be 120 though...)

Reply:
Ah I broke it up into a line of documentation and a line of regular comment, is that better?